src/shared/components/app/navbar.tsx

```diff
@@ -47,13 +49,18 @@ function handleLogOut(i: Navbar) {
 }
 
 export class Navbar extends Component<NavbarProps, NavbarState> {
+  private userAgent: UAParser.IResult = (
+    this.context.router.staticContext as RouterContext
+  ).userAgent;
+
   state: NavbarState = {
     unreadInboxCountRes: { state: "empty" },
     unreadReportCountRes: { state: "empty" },
     unreadApplicationCountRes: { state: "empty" },
   };
   collapseButtonRef = createRef<HTMLButtonElement>();
   mobileMenuRef = createRef<HTMLDivElement>();
+  isMobileUA = isMobile(this.userAgent);
```

Comment:
If we’re deciding to detect mobile this way –

It would be nice to have `isMobile` be a method with no parameters that can be reused in the future more easily throughout the app. That way, you can avoid having to add a `private` to this component and it would also make it much more of a utility method; It’s in `utils.ts` after all. ☺️

Also, it would be nice if we can avoid adding an additional NPM dependency.

Reply:
the argument here is inevitable since we need to read `this.context` where the user agent object lies which can be done only within the component

Reply:
I can rename the function though :)